e2e/test/MethodE2EMultiplexingOverAmqpTests.cs

```diff
@@ -228,65 +222,40 @@ public async Task Method_IoTHubSak_DeviceReceivesMethodAndResponseWithDefaultHan
-                {
-                    ConnectionStatus? lastConnectionStatus = null;
-                    ConnectionStatusChangeReason? lastConnectionStatusChangeReason = null;
-                    int setConnectionStatusChangesHandlerCount = 0;
-
-                    TestDevice testDevice = await TestDevice.GetTestDeviceAsync($"{DevicePrefix}_{i}_").ConfigureAwait(false);
-                    DeviceClient deviceClient = testDevice.CreateDeviceClient(transportSettings, authScope);
-                    deviceClients.Add(deviceClient);
-
-                    deviceClient.SetConnectionStatusChangesHandler((status, statusChangeReason) =>
-                    {
-                        setConnectionStatusChangesHandlerCount++;
-                        lastConnectionStatus = status;
-                        lastConnectionStatusChangeReason = statusChangeReason;
-                        _log.WriteLine($"{nameof(MethodE2EMultiplexingTests)}.{nameof(ConnectionStatusChangesHandler)}: status={status} statusChangeReason={statusChangeReason} count={setConnectionStatusChangesHandlerCount}");
-                        deviceClientConnectionStatusChangeCount[deviceClient] = setConnectionStatusChangesHandlerCount;
-                    });
+                _log.WriteLine($"{nameof(MethodE2EMultiplexingOverAmqpTests)}: Preparing to receive method for device {testDevice.Id}");
+                await MethodE2ETests.ServiceSendMethodAndVerifyResponse(
+                    testDevice.Id, MethodName, MethodE2ETests.DeviceResponseJson, MethodE2ETests.ServiceRequestJson).ConfigureAwait(false);
+            };
 
-                    Task methodReceivedTask = await setDeviceReceiveMethod(deviceClient).ConfigureAwait(false);
-                    await Task.WhenAll(
-                        MethodOperation.ServiceSendMethodAndVerifyResponse(testDevice.Id),
-                        methodReceivedTask).ConfigureAwait(false);
-                }
-            }
-            finally
+            Func<IList<DeviceClient>, Task> cleanupOperation = async (deviceClients) =>
             {
-                // Close and dispose all of the device client instances here
                 foreach (DeviceClient deviceClient in deviceClients)
                 {
-                    await deviceClient.CloseAsync().ConfigureAwait(false);
-
-                    // The connection status change count should be 2: connect (open) and disabled (close)
-                    Assert.IsTrue(deviceClientConnectionStatusChangeCount[deviceClient] == 2, $"Connection status change count for deviceClient {TestLogging.GetHashCode(deviceClient)} is {deviceClientConnectionStatusChangeCount[deviceClient]}");
-
-                    _log.WriteLine($"{nameof(MethodE2EMultiplexingTests)}: Disposing deviceClient {TestLogging.GetHashCode(deviceClient)}");
                     deviceClient.Dispose();
                 }
-            }
+                await Task.FromResult<bool>(false).ConfigureAwait(false);
```

Comment:
In general, return Task.CompletedTask to avoid another allocation.
Also, can't we just implement the template to account for null `cleanupOperation` and do nothing?